spark/v4.0/spark/src/test/java/org/apache/iceberg/spark/data/vectorized/parquet/TestParquetVectorizedReads.java

```diff
@@ -416,4 +439,60 @@ public void testUuidReads() throws Exception {
     }
     assertRecordsMatch(schema, numRows, data, dataFile, false, BATCH_SIZE);
   }
+
+  private void assertIdenticalFileContents(File actual, File expected, Schema schema)
+      throws IOException {
+    try (CloseableIterable<InternalRow> actualReader =
+        Parquet.read(Files.localInput(actual))
+            .project(schema)
+            .createReaderFunc(t -> SparkParquetReaders.buildReader(schema, t, ID_TO_CONSTANT))
+            .build()) {
+      Iterator<InternalRow> actualIterator = actualReader.iterator();
+      try (CloseableIterable<InternalRow> plainReader =
+          Parquet.read(Files.localInput(expected))
+              .project(schema)
+              .createReaderFunc(t -> SparkParquetReaders.buildReader(schema, t, ID_TO_CONSTANT))
+              .build()) {
+        Iterator<InternalRow> expectedIterator = plainReader.iterator();
+
+        List<InternalRow> expectedList = Lists.newArrayList();
+        expectedIterator.forEachRemaining(expectedList::add);
+        List<InternalRow> actualList = Lists.newArrayList();
+        actualIterator.forEachRemaining(actualList::add);
+
+        assertThat(actualList).isNotEmpty();
+        assertThat(actualList).hasSameSizeAs(expectedList);
+        assertThat(actualList).hasSameElementsAs(expectedList);
```

Comment:
Nit: Shall we include the `encoding` and `typeName` in the assertion message? That way, it’s easier to identify which pair failed if the test fails
```
assertThat(actualList)
    .as("Validation failed for encoding %s and type %s", encoding, typeName)
    .isNotEmpty()
    .hasSameSizeAs(expectedList)
    .hasSameElementsAs(expectedList);
```